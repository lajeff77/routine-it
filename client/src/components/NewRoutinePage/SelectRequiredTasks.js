import React, {useEffect, useState} from 'react';
import {Box, Card, Typography, Checkbox, FormGroup, FormControlLabel} from '@mui/material';
import {Formik, Form} from 'formik';

const SelectRequiredTasks = ({props}) => {
    const [checkedItems, setCheckedItems] = useState([]);

    let formik = props.formik;

    useEffect(() => {
        formik.validateForm();
    },[])

    useEffect(() => {
        switch(props.id){
            case 0:
                formik.setFieldValue("lowEffortRoutine", checkedItems);
                break;
            case 1:
                formik.setFieldValue("medEffortRoutine", checkedItems);                
                break;
            case 2:
                formik.setFieldValue("highEffortRoutine", checkedItems);
                break;
            default:
                //do nothing
                break;
        }
    }, [checkedItems]);

    const handleChange = (event) => {
        const { name, checked } = event.target;
        if (checked) {
            setCheckedItems(prevItems => [...prevItems, name]);
        } else {
            setCheckedItems(prevItems => prevItems.filter(item => item !== name));
        }
    };

    return(
        <Box sx={{margin:"20px", display:"flex", flexDirection:"column", alignItems:"center"}}>
            <Typography variant="h1" sx={{textAlign: 'center', margin: '30px'}}>{props.text.title}</Typography>
            <Card variant="outlined" sx={{align:"center", width:"70%", display:"flex", flexDirection:"column", outline:"2px"}}>
                <Typography variant="body1"  sx={{margin: '20px'}}>{props.text.prompt}</Typography>
                <Box>
                    <Formik initialValues={{list: checkedItems}} validateOnChange={true}>
                        <Form name="list">
                            {formik.values.tasks.map((item)=> {return(
                                <Box>
                                    <FormGroup sx={{margin:"20px"}}>
                                        <FormControlLabel control={<Checkbox name={item} checked={checkedItems.includes(item)} onChange={handleChange}/>} label={item} />
                                    </FormGroup>
                                </Box>)
                            })}
                        </Form>
                    </Formik> 
                 </Box>   
            </Card>
        </Box>
    )
}

export default SelectRequiredTasks;

