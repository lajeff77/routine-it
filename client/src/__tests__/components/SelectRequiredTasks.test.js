import { render } from "@testing-library/react";
import {expect, test} from '@jest/globals';
import React from "react";
import SelectRequiredTasks from "../../components/NewRoutinePage/SelectRequiredTasks";

const mockFormik = {
    values: {
        name: '',
        tasks: [],
        idealRoutine: [],
        averageRoutine: [],
        survivalRoutine: []
    },
    errors: {},
    touched: {},
    validateForm: jest.fn(),
    setFieldValue: jest.fn(),
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
};

const textMock = {
    title: '',
    prompt: ''
}

const testProps = {
    formik: mockFormik,
    list: [],
    text: textMock,
    id: 0
};

describe(SelectRequiredTasks.name, ()=>{
    test('SelectRequiredTasks matches snapshot', () => {
        const {container} = render(<SelectRequiredTasks props={testProps}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});