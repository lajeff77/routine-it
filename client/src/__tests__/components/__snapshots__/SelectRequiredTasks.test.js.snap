// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectRequiredTasks SelectRequiredTasks matches snapshot 1`] = `
<div
  class="MuiBox-root css-2u9sdq"
>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-3ns5jg-MuiTypography-root"
  />
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1yf7ndb-MuiPaper-root-MuiCard-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-altc5y-MuiTypography-root"
    />
    <div
      class="MuiBox-root css-0"
    >
      <form
        action="#"
        name="list"
      />
    </div>
  </div>
</div>
`;
